### PR TITLE
Offer

### PR DESCRIPTION
**Pull request by Bl4syo**
Created new model Offer

### CODE REVIEW

File: app/models/offer.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Offer < ApplicationRecord
+  validates :description, presence: true,
+                    length: { minimum: 5 }
```
Comment (zulixro) [inline app/models/offer.rb]: ok, looking good, but we need more validations. Validate presence of other attributes as well.

File: config/initializers/inflections.rb [hunk 2]
```diff
@@ -1,16 +1,16 @@
-# Be sure to restart your server when you modify this file.
```
Comment (zulixro) [inline config/initializers/inflections.rb]: You have some mysterious diffs: files you didn't really change and yet they still show up here. Just ignore them for now (or try to get rid of them in this branch and copy paste them from master branch - good activity to practice some branch switching).

File: db/migrate/20180521172343_create_offers.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class CreateOffers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :offers do |t|
+      t.references :product
```
Comment (zulixro) [inline db/migrate/20180521172343_create_offers.rb]: usually we first list the database columns like "description" or "title" and we add references at the end.
So:
create_table :offers do |t|
    t.string :title
    t.text :description
    t.integer :suggested_price
    t.string :status

    t.references :product
    t.references :customer

    t.timestamps
end

File: db/migrate/20180521172343_create_offers.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      t.text :description
+      t.string :title
+      t.integer :suggested_price
+      t.string :status
```
Comment (zulixro) [inline db/migrate/20180521172343_create_offers.rb]: You need to add default status (default: proposed)

File: db/migrate/20180521172343_create_offers.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+class CreateOffers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :offers do |t|
+      t.references :product
+      t.text :description
```
Comment (zulixro) [inline db/migrate/20180521172343_create_offers.rb]: If you add validations in model add them in migration as well. So you need to find out how to do it :)

File: db/migrate/20180521172343_create_offers.rb (hunk 3 above)
Comment (Bl4syo) [reply db/migrate/20180521172343_create_offers.rb]: Ok, modified accordingly

File: db/migrate/20180521172343_create_offers.rb (hunk 4 above)
Comment (Bl4syo) [reply db/migrate/20180521172343_create_offers.rb]: Done

File: db/migrate/20180521172343_create_offers.rb (hunk 5 above)
Comment (Bl4syo) [reply db/migrate/20180521172343_create_offers.rb]: Well, I have to keep looking for how to do that :)

File: app/models/offer.rb (hunk 1 above)
Comment (Bl4syo) [reply app/models/offer.rb]: Added more validation

File: app/models/offer.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Offer < ApplicationRecord
+  validates :description, presence: true, length: { minimum: 5 }
+  validates :title, presence: true
+  validates :product, presence: true
+  validates :suggested_price, numericality: { only_integer: true }
```
Comment (zulixro) [inline app/models/offer.rb]: what about `status`? You could add some validations for status as well.

File: app/models/offer.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+end
```
Comment (zulixro) [inline app/models/offer.rb]: it's time to add some associations. Add assosiation to product and to customer.

File: db/migrate/20180521172343_create_offers.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+class CreateOffers < ActiveRecord::Migration[5.1]
+  def change
+    create_table :offers do |t|
+      t.text :description
+      t.string :title
+      t.integer :suggested_price, :default => 0
+      t.string :status, :default => "Proposed"
```
Comment (zulixro) [inline db/migrate/20180521172343_create_offers.rb]: I would definitely use lowercase as a default value: `proposed` - someone can try to compare values or make a query to find records with a specific status value in the future - I think they would most probably look for all lowercase words as that is most common approach.

File: db/migrate/20180521172343_create_offers.rb (hunk 5 above)
Comment (zulixro) [reply db/migrate/20180521172343_create_offers.rb]: https://stackoverflow.com/questions/13122791/rails-validation-in-model-vs-migration?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

File: app/models/offer.rb (hunk 6 above)
Comment (Bl4syo) [reply app/models/offer.rb]: Added

File: app/models/offer.rb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+class Offer < ApplicationRecord
+  validates :description, presence: true, length: { minimum: 5 }
+  validates :title, presence: true
+  validates :product, presence: true
```
[the remark this answers is not in the document]
Comment (Bl4syo) [reply app/models/offer.rb]: Added

File: app/models/offer.rb (hunk 7 above)
Comment (Bl4syo) [reply app/models/offer.rb]: Added two associations, according to the specs on Trello "that offer will be owned by the customer and by the product". I think :)

File: db/migrate/20180521172343_create_offers.rb (hunk 8 above)
Comment (Bl4syo) [reply db/migrate/20180521172343_create_offers.rb]: Modified

File: app/models/offer.rb (hunk 7 above)
Comment (zulixro) [reply app/models/offer.rb]: Here's a hint: http://guides.rubyonrails.org/association_basics.html

**Pull request merged**